cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/security/trustedproxy/TrustedProxyAuthorizationService.java

```diff
@@ -22,30 +27,50 @@ public class TrustedProxyAuthorizationService extends AbstractLifeCycle implemen
 
   private final UserStore _serviceUserStore;
   private final Pattern _trustedProxyIpPattern;
+  private IdentityService _identityService;
 
-  TrustedProxyAuthorizationService(List<String> userNames, String trustedProxyIpPattern) {
+  protected TrustedProxyAuthorizationService(List<String> userNames, String trustedProxyIpPattern) {
     _serviceUserStore = new UserStore();
     userNames.forEach(u -> _serviceUserStore.addUser(u, SecurityUtils.NO_CREDENTIAL, new String[] { DefaultRoleSecurityProvider.ADMIN }));
     if (trustedProxyIpPattern != null) {
       _trustedProxyIpPattern = Pattern.compile(trustedProxyIpPattern);
     } else {
       _trustedProxyIpPattern = null;
     }
+    _identityService = new DefaultIdentityService();
   }
 
   @Override
-  public UserIdentity getUserIdentity(HttpServletRequest request, String name) {
+  public UserIdentity getUserIdentity(Request request, String name) {
     // ConfigurableSpnegoAuthenticator may pass names in servicename/host format but we only store the servicename
     int nameHostSeparatorIndex = name.indexOf('/');
     String serviceName = nameHostSeparatorIndex > 0 ? name.substring(0, nameHostSeparatorIndex) : name;
-    UserIdentity serviceIdentity = _serviceUserStore.getUserIdentity(serviceName);
+    UserPrincipal user = _serviceUserStore.getUserPrincipal(serviceName);
+    List<RolePrincipal> roles = _serviceUserStore.getRolePrincipals(serviceName);
```

Comment:
I'm just making a mental note here that `_serviceUserStore` is a bit confusing, perhaps in another PR we could change this to a less confusing name (and I know, I called it this way back then but it made sense for some reason 😄 ).